proposals/lip-update_block_schema_and_block_processing.md

```diff
@@ -0,0 +1,520 @@
+
+
+## Specification
+
+### Notation and Constants
+
+For the rest of this proposal we define the following constants.
+
+| Name          | Type    | Value       | Description |
+| ------------- |---------| ------------| ------------|
+| `MAX_PAYLOAD_SIZE_BYTES` | integer | TBD | The max size of a block payload in bytes.|
+| `MAX_ASSET_DATA_SIZE_BYTES` | integer | TBD | The max size of an assets entry in bytes.|
+|`SIGNATURE_LENGTH_BYTES`| integer | 64 | The length of a Ed25519 signature.|
+
+
+Furthermore, in the following we indicate with `block` be the block under consideration and with `previousBlock` the previous block of the chain.
+Calling a function `fct` from another module `module` is represented by `module.fct`.
+
+### Block
+
+#### JSON Schema
+
+Blocks are serialized and deserialized accordingly to the following JSON schema.
+
+```java
+blockSchema = {
+  "type": "object",
+  "required": ["header", "payload", "assets"],
+  "properties": {
+    "header": {
+      "dataType": "bytes",
+      "fieldNumber": 1
+    },
+    "payload": {
+      "type": "array",
+      "fieldNumber": 2,
+      "items": {
+        "dataType": "bytes"
+      }
+    },
+    "assets": {
+      "type": "array",
+      "fieldNumber": 3,
+      "items": {
+        "dataType": "bytes"
+      }
+    }
+  }
+}
+```
+
+#### Validation
+
+Blocks are validated at three different stages of their lifecycle.
+
+- **Static validation**: When a new block is received, some initial static checks are done to ensure that the serialized object follows the general structure of a block. 
+  These checks are performed immediately because they do not require access to the state store and can therefore be done very quickly.
+- **Block verification**: Properties that require access to the state store *before* the block has been executed are verified in this stage.
+In particular, these checks are performed *before* the state transitions implied by the modules are processed.
+- **Result verification**: In this stage we verification the properties that require access to the state store *after* the block has been executed, i.e. they can be verified only after the state transitions implied by the block execution have been performed.
+In particular, these checks are performed *after* the state transitions implied by the modules are processed.
+
+As part of the static validation checks, we check that the total size of the serialized transactions contained in the block payload is at most `MAX_PAYLOAD_SIZE_BYTES`.
+
+#### Block ID
+
+The block ID is calculated using the `blockID` function.
+This function returns a 32 bytes value or an error if the block header has an invalid signature format.
+
+```python
+blockID():
+  # Check that the signature length is 64 bytes
+  if length(block.header.signature) != SIGNATURE_LENGTH_BYTES:
+    return error
```

Comment:
I'm not sure why this check is done here. For me, this could be removed.

Reply:
I added a comment to the code to clarify the reasoning